src/thread_task.rs

```diff
@@ -323,24 +331,29 @@ unsafe fn create_filter_sbrow(
     let has_cdef = seq_hdr.cdef;
     let has_resize = (frame_hdr.size.width[0] != frame_hdr.size.width[1]) as c_int;
     let has_lr = f.lf.restore_planes;
-    let tasks = &mut *f.task_thread.tasks();
+    let tasks = &mut *fc.task_thread.tasks();
     let uses_2pass = (c.n_fc > 1 as c_uint) as c_int;
     let num_tasks = (f.sbh * (1 + uses_2pass)) as usize;
     tasks.grow_tasks(num_tasks);
     let task_idx = Rav1dTaskIndex::Task((f.sbh * (pass & 1)) as usize);
     if pass & 1 != 0 {
-        f.frame_thread_progress.entropy = AtomicI32::new(0);
+        fc.frame_thread_progress
+            .try_read()
+            .unwrap()
+            .entropy
+            .store(0, Ordering::Relaxed);
```

Comment:
It seems odd to have an atomic inside of an `RwLock`ed struct. Both give us interior mutability, and doing it this way means we're paying twice to mutate that field. Would it make sense to hoist the atomic field out of the `RwLock`?

Reply:
I could move the RwLock down to just the Vecs, we only need it to be able to resize the Vec.

Reply:
The relaxed atomic load/store is pretty much free, though, so I don't know if it's the biggest deal.